src/lib.rs

```diff
@@ -213,69 +204,64 @@ impl Default for BlendMode {
 }
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+#[serde(rename_all = "camelCase")]
+#[non_exhaustive]
 pub struct Layer {
     pub name: String,
     pub group: Option<String>,
     #[serde(default)] // 0 / missing for groups - editor shows "0" greyed out
     pub opacity: u32,
-    #[serde(rename = "blendMode", default)] // 0 / missing for groups - editor shows "Normal" greyed out
+    // 0 / missing for groups - editor shows "Normal" greyed out
     pub blend_mode: BlendMode,
     pub color: Option<Color>,
     pub data: Option<String>,
-
-    #[doc(hidden)] #[serde(skip)] pub _non_exhaustive: (), // more fields may be added
 }
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+#[serde(rename_all = "camelCase")]
+#[non_exhaustive]
 pub struct Slice {
     pub name: String,
     pub color: Color,
     pub data: Option<String>,
     pub keys: Vec<SliceKey>,
-
-    #[doc(hidden)] #[serde(skip)] pub _non_exhaustive: (), // more fields may be added
 }
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+#[serde(rename_all = "camelCase")]
+#[non_exhaustive]
 pub struct SliceKey {
     pub frame: u32,
     pub bounds: Rect,
     pub pivot: Option<Point>,
     pub center: Option<Rect>,
-
-    #[doc(hidden)] #[serde(skip)] pub _non_exhaustive: (), // more fields may be added
 }
 
-
 #[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+#[serde(rename_all = "camelCase")]
+#[non_exhaustive]
 pub struct Metadata {
     pub app: String,
     pub version: String,
     pub format: String,
     pub size: Dimensions,
     pub scale: String, // Surely this should be a number?
-    #[serde(default, rename = "frameTags")]
     pub frame_tags: Vec<Frametag>,
     #[serde(default)]
     pub layers: Vec<Layer>,
     pub image: Option<String>,
     #[serde(default)]
     pub slices: Vec<Slice>,
-
-    #[doc(hidden)] #[serde(skip)] pub _non_exhaustive: (), // more fields may be added
 }
 
-
 #[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+#[serde(rename_all = "camelCase")]
+#[non_exhaustive]
 pub struct SpritesheetData {
-    #[serde(deserialize_with = "deserialize_frames")]
```

Comment:
I’m just on my phone right now, so I can’t really check, but why can we remove this custom deserialization? Or just a mixup? ^^

Reply:
That's odd, definitely a mixup.

Reply:
But... aseprite itself doesn't belong to ggez at all :sweat_smile:
Perhaps we should change the repo name to something less misleading though.

Sure, it IS a useful project for other projects working on / using aseprite deserialization. But there are more out there than just this repo here. Searching for "aseprite" on crates.io gives me at least two other projects, one of them being https://crates.io/crates/asefile
Ok, they have their own test data, but still.

I'd argue that it doesn't _need_ to be part of ggez. It could be. But I don't see many reasons why we shouldn't just fork it and work with that. Sure, owning the original project is nice to update the crate, but why don't we just fork it under ggez until then and then offer to take over the crate (and merge back into the original project taking it over as well)?
That way we wouldn't have to wait, but and could still achieve all benefits, once the time comes and the owner agrees to give it to ggez. And if they don't want that its their good right and doesn't impede us much.

You can still test it by supplying a link to your forked repo in the Cargo.toml.

Reply:
That's true, actually I don't mind depending on the repo as is like now. The only annoying thing currently is that it has a lot of `_non_exhaustive_fields`, but for that I already created a PR. When that PR is merged quickly I'll update this PR, and when that one takes a lot of time we can see about forking it under ggez.